OSOE-844: Don't include a file called .htmlvalidate.json by default in Lombiq.UITestingToolbox (#361)
Lombiq/UI-Testing-Toolbox: Lombiq.Tests.UI/Docs/Configuration.md

### HTML validation configuration

> If you want to change some HTML validation rules from only a few specific tests, you can create a custom _.htmlvalidate.json_ file (e.g. _TestName.htmlvalidate.json_). This should extend the [default.htmlvalidate.json](../default.htmlvalidate.json) file (which is always copied into the build directory) by setting the value of `"extends"` to a relative path pointing to it. For example:

Reviewer: After this, also remove the "Make sure to also include the root attribute and set it to true" note below, which is not generally applicable.

```suggestion
If you want to change some HTML validation rules from only a few specific tests, you can create a custom _.htmlvalidate.json_ file (e.g. _TestName.htmlvalidate.json_). This should extend the [default.htmlvalidate.json](../default.htmlvalidate.json) file (which is always copied into the build directory) by setting the value of `"extends"` to a relative path pointing to it and declaring `"root": true`. For example:
```

```json
{
    "extends": [
        "./default.htmlvalidate.json"
    ],

    "rules": {
        "element-required-attributes": "off",
        "no-implicit-button-type": "off"
    },

>     "root":  true

Reviewer: There's an extra space after the colon. Please also fix in the files.

}
```

> Then you can change the configuration to use that:

Reviewer: ```suggestion
You can also create a completely standalone config file too, without any inheritance, but we'd recommend against that. 

You can change the configuration to use the above file as follows:
```

```cs
changeConfiguration: configuration => configuration.HtmlValidationConfiguration.HtmlValidationOptions =
    configuration.HtmlValidationConfiguration.HtmlValidationOptions

>         .CloneWith(validationOptions => validationOptions.ConfigPath =

Reviewer: Why do we actually need to clone? I think simply setting the path should be enough.

Author: I don't know, it was already like this in the Configuration.md so I assumed it's necessary.
@DemeSzabolcs Do you remember the reason or where it came from? Since you added the docs in #333.

Reviewer: FYI I implemented a non-cloning way to set the path under https://github.com/Lombiq/UI-Testing-Toolbox/pull/362 too (will remove it) and nothing bad happened. The config object is unique to the test (ultimately due to `OrchardCoreUITestExecutorConfiguration` being instantiated for each test in `OrchardCoreUITestBase`) so changing its state isn't a problem.

Reviewer 2: > I don't know, it was already like this in the Configuration.md so I assumed it's necessary. @DemeSzabolcs Do you remember the reason or where it came from? Since you added the docs in #333.

I did that based on this:
https://github.com/atata-framework/atata-htmlvalidation/issues/8#issuecomment-1864839487

But yeah, `CloneWith` might not needed.

Author: Thanks for the context! In this case it's indeed not necessary because the example was using `HtmlValidationOptions.Default`, but we use `new HtmlValidationOptions { ... }` so it's all unique instances.

            Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestName.htmlvalidate.json")));
```

Though if the file is in the base directory like above, then it can be simplified using the `WithRelativeConfigPath(params string[] pathSegments)` method:

```cs
changeConfiguration: configuration => configuration.HtmlValidationConfiguration.WithRelativeConfigPath("TestName.htmlvalidate.json");
```

> Make sure to also include the `root` attribute and set it to `true` inside the custom _.htmlvalidate.json_ file and include it in the test project like this:

Reviewer: ```suggestion
If you want to do this for all tests in the project, just put an _.htmlvalidate.json_ file into the project root and it will be picked up without further configuration.

Make sure to also include the `root` attribute and set it to `true` inside the custom _.htmlvalidate.json_ file and include it in the test project like this:
```
